catalog/core/catalog-core-api-impl/src/test/groovy/ddf/catalog/util/impl/ResultIterableSpec.groovy

```diff
@@ -195,31 +197,98 @@ class QueryResultPaginatorSpec extends Specification {
         1.upto(resultSize + 1, { results << resultIterator.next() })
 
         then:
-        1.upto(resultSize + 1, { results.get(it - 1).distanceInMeters == (Double) it })
+        1.upto(resultSize + 1, { assert results.get(it - 1).distanceInMeters == (Double) it })
+    }
+
+    def "Properly pages when default page size is used with no maxResults"() {
```

Comment:
this unit test can be removed since it is mostly a duplicate of the more thorough one below